dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/DSpaceObjectMetadataRemoveOperation.java

```diff
@@ -66,11 +72,13 @@ public R perform(Context context, R resource, Operation operation) throws SQLExc
     private void remove(Context context, DSpaceObject dso, DSpaceObjectService dsoService, MetadataField metadataField,
                         String index) {
         metadataPatchUtils.checkMetadataFieldNotNull(metadataField);
+        String msg;
         try {
             if (index == null) {
                 // remove all metadata of this type
                 dsoService.clearMetadata(context, dso, metadataField.getMetadataSchema().getName(),
                         metadataField.getElement(), metadataField.getQualifier(), Item.ANY);
+                provenanceService.removeMetadata(context, dso, metadataField);
```

Comment:
why not in dsoService?

Reply:
No, because it doesn’t always make sense to add provenance when the clearMetadata method is called...

Reply:
please, give me a reason

Reply:
The reason:
This `dsoService `can be used for any `DSpaceObject`, such as `EPerson`, `Community`, etc. If we add a provenance service to this method, it will add a provenance metadata for every `DSpaceObject` on which this method is called.

Reply:
The `provenanceService.removeMetadata(context, dso, metadataField);` could be really used in the `bitstreamService.clearMetadata` what is an implementation for the dsoService.